### DISCUSSION TITLE
How to improve the ADC precision?

### DISCUSSION THREAD

**Discussion by zhujt4github** [stm32duino/Arduino_Core_STM32#1501] [Q&A] [answered]
The default precision of stm32G0 is 12 bits. How can I increase the precision to 13 bits or higher?

**Comment by fpistm** [comment, marked as answer]
Hi, 
simply use `analogreadresolution()`:
https://www.arduino.cc/reference/en/language/functions/zero-due-mkr-family/analogreadresolution/


Anyway G0 max ADC resolution is 12 bits so increasing the resolution will simply map the resolution which will not be very precise.

**Reply by zhujt4github** [reply]
Thanks for your reply, 
In the stm32g0 manual I found this below:

> A native 12-bit analog-to-digital converter is embedded into STM32G030x6/x8 devices. **It
can be extended to 16-bit resolution through hardware oversampling.**

Do we have this feature or similar features? If not how can I achieve this feature.

**Reply by fpistm** [reply]
No oversampling is not managed.
If you want achieve this then you will have to use HAL to configure and use ADC with the oversampling.

**Reply by zhujt4github** [reply]
Got it, thank you very much!
